packages/grafana-ui/src/components/Table/utils.ts

```diff
@@ -533,3 +534,14 @@ export function getAlignmentFactor(
     return alignmentFactor;
   }
 }
+
+export function hasTimeField(data: DataFrame): boolean {
+  return data.fields.some((field) => field.type === FieldType.time);
+}
+
+// since the conversion from timeseries panel crosshair to time is pixel based, we need
+// to set a threshold where the table row highlights when the crosshair is within a certain point
+// because multiple pixels (converted to times) may represent the same point/row in table
+export function isPointTimeValAroundTableTimeVal(pointTime: number, rowTime: number) {
+  return Math.abs(Math.floor(pointTime) - rowTime) < POINT_THRESHOLD_MS;
```

Comment:
I think we need some other kind of method of finding this number. When looking at a big time range you need to get lucky to get this to match. Could the threshold adapt to the time range?

![Peek 2023-11-20 16-06](https://github.com/grafana/grafana/assets/468940/098862cd-18c2-4e88-b02d-366cd8e2fe2f)

Reply:
Indeed, need to look more into that

Reply:
I've added an adaptive threshold that should change on data range changes. WDYT?